### PR TITLE
Fix Codex reasoning leaking into final assistant message

### PR DESCRIPTION
**Pull request by Th0rgal**
## Summary
- fix Codex message content parsing to ignore `reasoning`/`thinking` blocks when extracting assistant text
- keep thinking streamed via dedicated `thinking` events while final assistant message only contains answer text
- add regression test for mixed content arrays to prevent future leakage

## Root cause
Codex `message` items can include mixed content blocks (reasoning + output text). The parser concatenated all text-like blocks from `content[]`, so reasoning text was appended into `TextDelta`, then persisted into final `assistant_message`.

## Validation
- `cargo test -q extract_text_field_from_content_array -- --nocapture`
- deployed to dev backend (`sandboxed-sh-dev`) and verified health
- mission API verification on dev:
  - mission: `fc9a7c39-7116-466c-a6f1-8310d65ad635`
  - streamed events include:
    - `thinking`: `**Formulating a concise response**`
    - `assistant_message`: `2 plus 2 equals 4.`
  - final assistant message no longer includes thinking text

---

> [!NOTE]
> **Medium Risk**
> Touches streaming/parsing and mission completion/timeout logic, which can affect when runs finish or are killed; behavior changes are covered by targeted new tests but could still impact edge-case provider event sequences.
> 
> **Overview**
> **Fixes Codex output parsing** so `extract_text_field` skips `reasoning`/`thinking` blocks inside mixed `content[]`, while still emitting those blocks via dedicated thinking events (via `extract_text_field_with_reasoning`). Adds regression tests to ensure mixed content arrays don’t leak reasoning into assistant text.
> 
> **Adjusts OpenCode mission streaming/health behavior** by treating `response.incomplete` as non-terminal (wait for `response.completed`/idle signals) and adding separate heartbeat tracking so the global inactivity kill uses a heartbeat-only grace period (120s normal timeout, up to 420s if recent heartbeats). Adds SSE parsing tests for `response.incomplete` vs `response.completed` terminal behavior.
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit e6a2685c8bbef3a9675d17257269e152f4f49b8a. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

### CODE REVIEW

File: src/backend/codex/mod.rs [hunk 1]
```diff
@@ -548,6 +548,10 @@ fn extract_text_field(
         let mut out = String::new();
         let items = value.as_array()?;
         for item in items {
+            let item_type = item.get("type").and_then(|v| v.as_str()).unwrap_or("");
+            if matches!(item_type, "reasoning" | "thinking") {
+                continue;
+            }
```
[the remark this answers is not in the document]
Comment (Th0rgal) [reply src/backend/codex/mod.rs:570]: Fixed in 2f6962de: reasoning/thinking items now use extract_text_field_with_reasoning, while message extraction still filters reasoning blocks.
